stetho/src/main/java/com/facebook/stetho/inspector/MethodDispatcher.java

```diff
@@ -63,6 +64,11 @@ public JSONObject dispatch(JsonRpcPeer peer, String methodName, @Nullable JSONOb
           "Not implemented: " + methodName,
           null /* data */));
     }
+
+    Stetho.getAnalyticsLogger().track(
+        "inspector_method_called",
+        Collections.singletonMap("name", methodName));
```

Comment:
`Map` (or more accurately `HashMap`) is a very slow class to reach for in Java.  It scales well for large values of n but has incredibly high fixed overhead so that a large number of instances where n is small can have a staggeringly negative effect on overall memory usage and app performance.  Google offered an alternative, `SimpleArrayMap` which uses far less memory and trades the `O(1)` lookup time for `O(log n)` (which is faster where n<5) but in your case since you actually just want to iterate a key/value pair you should reach for a simple key/value list.

I'd recommend either defining your own simple class for this, with syntax similar to how you're using `Collections.singletonMap`:

```java
analyticsLogger.track(
    "inspector_method_called", 
    KeyValues.of("name", methodName));
```

or just taking advantage of the fact that a String[] with an even number of items is essentially a key/value list of alternating keys and values:

```java
analyticsLogger.track(
    "inspector_method_called",
    "name", methodName,
    "someOtherThing", bla,
    ...);
```

I'd be happy to discuss specifics with you in FB4A and Messenger about just how much RAM we use in map overhead :)